docs/plugin-architecture.md

```diff
@@ -0,0 +1,307 @@
+# Plugin Architecture Documentation
+
+This document provides a comprehensive overview of the WooCommerce PayPal Payments plugin architecture, explaining its modular design and how the various components work together.
+
+## Overview
+
+The WooCommerce PayPal Payments plugin is built using a modular architecture powered by the **Inpsyde Modularity framework**. This design provides:
+
+- **Modular Structure**: Each feature is contained within its own module with clear boundaries
+- **Dependency Injection**: PSR-11 container for service management and dependency resolution
+- **Feature Flags**: Dynamic module loading based on environment variables and filters
+- **Extensibility**: Well-defined extension points for customization and enhancement
+- **Maintainability**: Clear separation of concerns and consistent patterns
+
+## Core Components
+
+### Main Plugin File
+
+The plugin initialization begins in `woocommerce-paypal-payments.php:40-45`, which:
```

Comment:
I would recommend to not use any line numbers here in docs, cause with every change, we would also need to update docs
```suggestion
The plugin initialization begins in `woocommerce-paypal-payments.php`, which:
```